src/internal/template/template.go

```diff
@@ -86,6 +86,11 @@ func (values Values) Apply(component types.ZarfComponent, path string) {
 		"GIT_AUTH_PULL":      values.secret.gitPull,
 	}
 
+	// Include the data injection marker template if the component has data injections
+	if len(component.DataInjections) > 0 {
```

Comment:
No real reason but consistency, there is a scenario where someone might try to apply a data injection to a different component's manifests, but it wouldn't be a good idea since we change the helm wait logic based on if the current component has a data injection.

Reply:
Ok yeah I am fine with it just wanted to make sure there wasn't something I was missing.  It would only affect users in niche ways anyway.